internal/pkg/service/buffer/storage/local/writer/disksync/disksync.go

```diff
@@ -117,168 +117,141 @@ func NewSyncer(logger log.Logger, clock clock.Clock, config Config, chain chain)
 	return w
 }
 
-// Write writes to the underlying writer and trigger synchronization when the configured Config.BytesTrigger volume is exceeded.
-func (s *Syncer) Write(p []byte) (n int, err error) {
-	if err := s.ctx.Err(); err != nil {
-		return 0, errors.Errorf(`syncer is closed: %w`, err)
+// AddWriteOp increments number of high-level writer operations,
+// for example writing one row of the table is one high-level write operation.
+func (s *Syncer) AddWriteOp(n uint) {
+	s.writeOpsCount.Add(uint64(n))
+}
+
+// DoWithNotify provides wrapping for multiple write operations and waiting for them to be synced to disk.
+// This is ensured by shared notifierLock, so notifier cannot be swapped during the method,
+// but parallel writes are not blocked. The lock blocks the TriggerSync method,
+// so operations are expected to be short.
+func (s *Syncer) DoWithNotify(do func() error) (notifier *notify.Notifier, err error) {
+	// Get notifier and block it change during write, see doSync method
+	// Note: *notify.Notifier(nil).Wait() is a valid call
+	if s.config.Wait {
+		s.notifierLock.RLock()
+		defer s.notifierLock.RUnlock()
+		notifier = s.notifier
 	}
 
-	// Write data to the underlying writer
-	n, err = s.writer.Write(p)
-	if err != nil {
-		return n, err
+	if err = do(); err != nil {
+		return nil, err
 	}
 
-	s.onWrite(n)
-	return n, nil
+	return notifier, nil
 }
 
-// WriteString - see Write,
-// some writes are optimized for writing strings
-// without an unnecessary conversion from []byte to string,
-// so both methods are supported.
-func (s *Syncer) WriteString(str string) (n int, err error) {
-	if err := s.ctx.Err(); err != nil {
-		return 0, errors.Errorf(`syncer is closed: %w`, err)
+// WriteWithNotify writes to the underlying writer.
+// Returned *notify.Notifier can be used to wait for disk sync.
+func (s *Syncer) WriteWithNotify(p []byte) (n int, notifier *notify.Notifier, err error) {
```

Comment:
Pridal som tu aj `WriteWithNotify`, ako si navrhoval, dobré sa to používa v testoch, ale opat, nejde to použit priamo s CSV Writer:

CSV Writer si totiz riesi volanie `syncer.Write` sam (musi tam byt standardny interface `io.Writer`), preto to potom ako pseudo kod vyzera takto:

```
w := csv.NewWriter(syncer(chain))

notifier := syncer.DoWithNotify(func() {
    w.WriteRow([]string{....})
})

return notifier.Wait()
```

^^^ CSV Writer nemusel `syncer.Writer` este zavolat ani raz, ale mohol.
Urcite vsak vsetko zapise vramci `Sync` operacie, kde sa vola `w.Flush()`.